python/dolfinx/wrappers/mesh.cpp

```diff
@@ -167,6 +167,28 @@ void declare_mesh(nb::module_& m, std::string type)
   std::string pyclass_geometry_name = std::string("Geometry_") + type;
   nb::class_<dolfinx::mesh::Geometry<T>>(m, pyclass_geometry_name.c_str(),
                                          "Geometry object")
+      .def(
+          "__init__",
+          [](dolfinx::mesh::Geometry<T>* self,
+             std::shared_ptr<const dolfinx::common::IndexMap> index_map,
+             nb::ndarray<std::int32_t, nb::ndim<1>, nb::c_contig> dofmap,
```

Comment:
I think this would make more sense from Python as a 2D array.

Reply:
I don't agree, as we would have to pad it in the nanobind-wrapper, as we have forced input as three-dimensional in the Geometry constructor in C++.

Reply:
I don't follow. It's naturally a 2D array. See https://github.com/FEniCS/dolfinx/blob/7bb01adaa736a325ca048e37aec0b51cf9ff1f2e/cpp/dolfinx/mesh/Geometry.h#L121.

Why would it need padding?

Reply:
Sorry, wrote this in the wrong place, meant for the points. I've now modified the code to take in 2D arrays for dofmap and points, but kept the `dim` argument.